shared_crates/package/src/manifest.rs

```diff
@@ -73,6 +74,64 @@ fn return_true() -> bool {
     true
 }
 
+#[derive(
+    Clone,
+    Debug,
+    PartialEq,
+    parse_display::Display,
+    parse_display::FromStr,
+    serde_with::SerializeDisplay,
+    serde_with::DeserializeFromStr,
+)]
+pub enum AmbientRuntimeVersion {
+    #[display("{major}.{minor}.{patch}")]
+    Stable { major: u32, minor: u32, patch: u32 },
```

Comment:
You might want to use `Version` here instead? You can then reuse its logic for semver etc

Reply:
@philpax Hm any reason we're not using https://crates.io/crates/semver instead of our own?

Reply:
Oh, I didn't see this - the reason was to provide deterministic behaviour for comparisons, especially with our `-dev`/`-alpha`/etc editions, and it grew out of the existing data structure. It looks like `semver` should cover this, so it shouldn't be too problematic to replace it.